src/views/NotificationPreferenceModal.vue

```diff
@@ -0,0 +1,209 @@
+      return prefs
+    }, {})
+    this.initialNotificationPrefState = JSON.parse(JSON.stringify(this.notificationPrefState))
+  },
+  methods: {
+    closeModal() {
+      modalController.dismiss({ dismissed: true });
+    },
+    toggleNotificationPref(enumId: string, value: boolean) {
+      // updates the notificationPrefToUpate to check which pref
+      // values were updated from their initial values
+      if (value !== this.notificationPrefState[enumId]) {
+        // updating this.initialNotificationPref as it is used to
+        // determine the save button disable state, hence, updating
+        // is necessary to recompute isButtonDisabled property
+        value
+          ? (this.notificationPrefToUpate.subscribe.push(enumId), this.notificationPrefState[enumId] = true)
+          : (this.notificationPrefToUpate.unsubscribe.push(enumId), this.notificationPrefState[enumId] = false)
+      } else {
+        !value
+          ? this.notificationPrefToUpate.subscribe.splice(this.notificationPrefToUpate.subscribe.indexOf(enumId), 1)
+          : this.notificationPrefToUpate.unsubscribe.splice(this.notificationPrefToUpate.subscribe.indexOf(enumId), 1)
+      }
+    },
+    async updateNotificationPref() {
+      // TODO disbale button if initial and final are same
+      // added loader as the API call is in pending state for too long, blocking the flow
+      emitter.emit("presentLoader");
+      try {
+        await this.handleTopicSubscription()
+      } catch (error) {
+        console.error(error)
+      } finally {
+        emitter.emit("dismissLoader")
+      }
+    },
+    async handleTopicSubscription() {
+      const subscribeRequests = [] as any
+      const oms = this.instanceUrl
+      const facilityId = (this.currentFacility as any).facilityId
+      this.notificationPrefToUpate.subscribe.map(async (enumId: string) => {
+        const topicName = generateTopicName(oms, facilityId, enumId)
+        await subscribeRequests.push(subscribeTopic(topicName, process.env.VUE_APP_NOTIF_APP_ID).catch((err: any) => {
+          return err;
+        }))
+      })
+
+      const unsubscribeRequests = [] as any
+      this.notificationPrefToUpate.unsubscribe.map(async (enumId: string) => {
+        const topicName = generateTopicName(oms, facilityId, enumId)
+        await unsubscribeRequests.push(unsubscribeTopic(topicName, process.env.VUE_APP_NOTIF_APP_ID).catch((err: any) => {
+          return err;
+        }))
+      })
+
+      const responses = await Promise.all([...subscribeRequests, ...unsubscribeRequests])
+      const successCount = responses.reduce((successCount: number, response: any) => {
+        if (response.successMessage) {
+          successCount++
+        }
+        return successCount
+      }, 0)
+
+      // using successCount count to handle toast message
+      if (successCount === this.notificationPrefToUpate.subscribe.length + this.notificationPrefToUpate.unsubscribe.length) {
```

Comment:
We should think on another way to check if any of the request failed instead of checking for count.